### PR TITLE
Bug 6890: throw an error when sending service emails for docket entries without an index

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

**Review by kkoskelin — APPROVED** [review]
An excellent strategy for catching this.  Nice!

File: shared/src/business/entities/cases/Case.js [hunk 1]
```diff
@@ -1197,7 +1197,7 @@ Case.prototype.generateNextDocketRecordIndex = function () {
 /**
  *
  * @param {DocketEntry} updatedDocketEntry the docket entry to update on the case
- * @returns {Case} the updated case entity
+ * @returns {DocketEntry} the updated DocketEntry entity
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/Case.js]: 😮

Comment (rachaelparris) [reply shared/src/business/entities/cases/Case.js]: Whoops, I went a different direction. I'll put this back.

File: shared/src/business/useCases/trialSessions/setNoticesForCalendaredTrialSessionInteractor.js [hunk 2]
```diff
@@ -298,8 +300,6 @@ exports.setNoticesForCalendaredTrialSessionInteractor = async ({
           servedParties,
         });
     }
-
-    return servedParties;
```
Comment (kkoskelin) [inline shared/src/business/useCases/trialSessions/setNoticesForCalendaredTrialSessionInteractor.js:302]: If removing the return value, should we update jsdoc?

**Review by kkoskelin — APPROVED** [review]
This stuff was everywhere... wow!  Nice work.
I won't hold up the PR for the jsdoc item.

**Pull request merged**